crates/bitwarden/src/tool/generators/password.rs

```diff
@@ -40,8 +41,85 @@ pub struct PassphraseGeneratorRequest {
     pub include_number: Option<bool>,
 }
 
-pub(super) fn password(_input: PasswordGeneratorRequest) -> Result<String> {
-    Ok("pa11w0rd".to_string())
+const DEFAULT_PASSWORD_LENGTH: u8 = 16;
+
+const UPPER_CHARS_AMBIGUOUS: [char; 2] = ['I', 'O'];
+const UPPER_CHARS: [char; 24] = [
+    'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'J', 'K', 'L', 'M', 'N', 'P', 'Q', 'R', 'S', 'T', 'U',
+    'V', 'W', 'X', 'Y', 'Z',
+];
+const LOWER_CHARS_AMBIGUOUS: [char; 1] = ['l'];
+const LOWER_CHARS: [char; 25] = [
+    'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't',
+    'u', 'v', 'w', 'x', 'y', 'z',
+];
+const NUMBER_CHARS_AMBIGUOUS: [char; 2] = ['0', '1'];
+const NUMBER_CHARS: [char; 8] = ['2', '3', '4', '5', '6', '7', '8', '9'];
+const SPECIAL_CHARS: [char; 8] = ['!', '@', '#', '$', '%', '^', '&', '*'];
+
+pub(super) fn password(input: PasswordGeneratorRequest) -> Result<String> {
+    // Generate all character dictionaries
+    fn gen_chars(chars: &[char], ambiguous: &[char], avoid_ambiguous: bool) -> Vec<char> {
+        let mut chars = chars.to_vec();
+        if !avoid_ambiguous {
+            chars.extend_from_slice(ambiguous);
+        }
+        chars
+    }
+    let avoid_ambiguous = input.avoid_ambiguous.unwrap_or(false);
+    let lower_chars = gen_chars(&LOWER_CHARS, &LOWER_CHARS_AMBIGUOUS, avoid_ambiguous);
+    let upper_chars = gen_chars(&UPPER_CHARS, &UPPER_CHARS_AMBIGUOUS, avoid_ambiguous);
+    let number_chars = gen_chars(&NUMBER_CHARS, &NUMBER_CHARS_AMBIGUOUS, avoid_ambiguous);
+    let all_chars = lower_chars
+        .iter()
+        .chain(&upper_chars)
+        .chain(&number_chars)
+        .chain(&SPECIAL_CHARS)
+        .collect::<Vec<_>>();
```

Comment:
Would it make sense to extract this to a function and test it to ensure the correct chars are available?